src/config.h

```diff
@@ -0,0 +1,134 @@
+#pragma once
+
+#include <cstdint>
+#include <iostream>
+#include <map>
+#include <variant>
+
+#include "types.h"
+
+namespace bpftrace {
+
+enum class ConfigSource
+{
+  default_,
+  env_var,
+};
+
+enum class ConfigKey
+{
+  ast_max_nodes,
+  cat_bytes_max,
+  debug_output,
+  no_cpp_demangle,
+  log_size,
+  map_keys_max,
+  max_probes,
+  max_bpf_progs,
+  max_type_res_iterations,
+  perf_rb_pages,
+  stack_mode,
+  strlen,
+  str_trunc_trailer,
+  user_symbol_cache_type,
+  verify_llvm_ir,
+};
+
+struct ConfigValue
+{
+  ConfigSource source;
```

Comment:
Maybe:

```suggestion
  ConfigSource source = ConfigSource::default_;
```

and then in the default map below you can do:

```
    { ConfigKey::ast_max_nodes, { .value = (uint64_t)0 } },
```

Reply:
I like it!